### PR TITLE
fix: [#4470] Migrate off @azure/ms-rest-js botbuilder-ai

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR replaces the uses of **_@azure/ms-rest-js_** in the following botbuilder-js libraries.

- [ ] botbuilder
- [x] botbuilder-ai
- [ ] botframework-connector
- [ ] tool

## Specific Changes
  - Replaced the **_@azure/ms-rest-js_** imports with **_@azure/core-http_**.
  - Added **_@azure/core-http_** to libraries package.json.
  - Added **_@azure/core-util_** version 1.5.0 to resolutions.
  - Updated unit tests to work with core-http **_RestError_**.

## Testing
The following image shows some samples working after the update.
image
image

### CODE REVIEW

File: libraries/botbuilder-ai/src/luisRuntimeClientContext.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
+ */
```
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/luisRuntimeClientContext.ts:8]: I think we should remove this comment. It's not an autogenerated code.

File: libraries/botbuilder-ai/src/luisV2-models/sentiment.ts [hunk 2]
```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ *
+ * Code generated by Microsoft (R) AutoRest Code Generator.
+ * Changes may cause incorrect behavior and will be lost if the code is regenerated.
+ */
```
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/luisV2-models/sentiment.ts:8]: same here.

File: libraries/botbuilder-ai/src/luisV2-models/sentiment.ts
Comment (ceciliaavila) [inline libraries/botbuilder-ai/src/luisV2-models/sentiment.ts:1]: Is correct to name this file sentiment? It contains a lot of other interfaces.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/botbuilder-ai/src/luisV2-models/sentiment.ts
Comment (JhontSouth) [reply libraries/botbuilder-ai/src/luisV2-models/sentiment.ts:1]: I'm extracting the logic and name from _@azure/cognitiveservices-luis-runtime_

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /4577

**Pull request closed without merging**